source/dofs/dof_handler_policy.cc

```diff
@@ -3037,10 +3041,14 @@ namespace internal
             &tree_indices
             &dof_numbers_and_indices;
 
-            quadrants.resize (quadrants_as_chars.size() / sizeof(quadrants[0]));
-            std::memcpy(quadrants.data(),
-                        quadrants_as_chars.data(),
-                        quadrants_as_chars.size());
+            if (quadrants_as_chars.size()>0)
+              {
+                Assert(quadrants.data() != nullptr, ExcInternalError());
+                quadrants.resize (quadrants_as_chars.size() / sizeof(quadrants[0]));
+                std::memcpy(quadrants.data(),
+                            quadrants_as_chars.data(),
+                            quadrants_as_chars.size());
+              }
```

Comment:
you might be missing an
```
  else
    quadrants.clear();
```
here to restore the functionality of the previous code in case `quadrants_as_chars.size()==0`.

Reply:
Thanks! I added it.